etna/records/models.py

```diff
@@ -447,12 +460,196 @@ def get_datalayer_data(self, request: HttpRequest) -> Dict[str, Any]:
+                    facilities=facilities,
+                    comments=mark_safe(document("comments").text()),
+                )
+        return further_info
+
+    def _get_trasformed_archive_record_creators_info(
+        self, collection_name
+    ) -> Tuple(CollectionInfo, None):
+        """
+        returns transformed record creators info
+        """
+        collection_info = None
+        if archive_links := self.get("links", ()):
+            filtered_values = find_all(
+                archive_links,
+                predicate=lambda i: i["identifier"][0]["value"]
+                == collection_name.get("api_links_indentifier_value"),
+            )
+            info_list = []
+            for item in filtered_values:
+                info_list.append(
+                    {
+                        "summary_title": item.get("summary", {}).get("title", ""),
+                        "url": reverse(
+                            "details-page-machine-readable",
+                            kwargs={"iaid": item.get("@admin", {}).get("id", "")},
+                        ),
+                        "place": item.get("place", {})
+                        .get("name", [{}])[0]
+                        .get("value", ""),
+                    }
+                )
+            if info_list:
+                collection_info = CollectionInfo(
+                    name=collection_name.get("name"),
+                    display_name=collection_name.get("display_name"),
+                    long_display_name=collection_name.get("long_display_name"),
+                    count=len(info_list),
+                    info_list=info_list,
+                )
+
+        return collection_info
+
+    def _get_transformed_archive_nra_records_info(
+        self, collection_name
+    ) -> Tuple(CollectionInfo, None):
+        """
+        returns transformed nra records info
+        """
+        collection_info = None
+        if manifestations := self.get("manifestations", ()):
+            info_list = []
+            for item in manifestations:
+                info_list.append(
+                    {
+                        "identifier_title": f'NRA {item.get("identifier",[{}])[0].get("value", "") } {item.get("title", [{}])[0].get("value", "")}',
+                        "url": item.get("url", ""),
+                    }
+                )
+            if info_list:
+                collection_info = CollectionInfo(
+                    name=collection_name.get("name"),
+                    display_name=collection_name.get("display_name"),
+                    long_display_name=collection_name.get("long_display_name"),
+                    count=len(info_list),
+                    info_list=info_list,
+                )
+
+        return collection_info
+
+    @cached_property
+    def archive_collections(self) -> ArchiveCollections:
+        """
+        returns archive collection info for record creators and nra records
+        """
```

Comment:
Is there any reason why these two data structures need to be collected into a single object/value in this way? Might it be simpler to just access each value separately in the template?

Reply:
The Record Creators and NRA records are from different sources but are the under  Collections - in terms according to the [api ticket](https://national-archives.atlassian.net/browse/DF-537) . Also NRA records paper_catalogue, appears under Summary in the static html prototype. Its fields output is "somewhat similar" to NRA in terms of presentation. So due to these similarities collected into one object.